src/npg_porch_cli/api_cli_user.py

```diff
@@ -0,0 +1,112 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+
+from npg_porch_cli.api import PorchRequest
+
+
+def run():
+    """
+    Parses command line arguments, send the request to porch server API,
+    prints out the server's response to STDOUT.
+
+    The text below assumes that this code is deployed a a script called
+    `npg_porch_client`.
+
+    One positional and one names argument to teh script are required.
+    The positional argument is the name of the action to be performed.
+
+      npg_porch_client list_pipeline --baseurl SOME_URL
+
+    A full list of actions:
+        list_tasks
+        list_pipeline
+        add_pipeline
+        add_task
+        claim_task
+        update_task
+
+    Though most of named arguments are optional, some actions require
+    certain combinations of arguments to be defined.
+
+    All list actions do not require any optional arguments defined. If
+    `--pipeline_name` is defined, `list_tasks` returns a list of tasks for
+    this pipeline, otherwise all registered tasks are returned.
+
+    All non-list actions require all `--pipeline`, `pipeline_url` and
+    `--pipeline_version` defined.
+
+    The `add_task` and `update_task` actions require the `--task_json` to
+    be defined. In addition to this, for the `update_task` action `--status`
+    should be defined.
+
+    NPG_PORCH_TOKEN environment variable should be set up to teh value of
+    either an admin or project-specific token.
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="npg_porch_client",
+        description="npg_porch (Pipeline Orchestration)API Client",
+        epilog="The server JSON reply is printed to STDOUT",
+    )
+
+    parser.add_argument(
+        "action",
+        type=str,
+        help="Action to send to npg_porch server API",
+        choices=[
+            "list_tasks",
+            "list_pipeline",
+            "add_pipeline",
+            "add_task",
+            "claim_task",
+            "update_task",
+        ],
+    )
+    parser.add_argument("--baseurl", type=str, required=True, help="Base URL")
+    parser.add_argument(
+        "--certfile", type=str, help="Server CA certificate path, optional"
+    )
+    parser.add_argument(
+        "--pipeline_url", type=str, help="Pipeline git project URL, optional"
+    )
+    parser.add_argument(
+        "--pipeline_version", type=str, help="Pipeline version, optional"
+    )
+    parser.add_argument("--pipeline", type=str, help="Pipeline name, optional")
+    parser.add_argument("--task_json", type=str, help="Task as JSON, optional")
+    parser.add_argument("--status", type=str, help="New status to set, optional")
```

Comment:
Might be able to constrain the status to set here, as for the actions listed above? Might not be better than relying on the client API module.

Reply:
Statuses are now retrieved dynamically from teh OpenAPI schema